src/MySqlConnector/ColumnReaders/IColumnReader.cs

```diff
@@ -0,0 +1,9 @@
+namespace MySqlConnector.ColumnReaders;
+using MySqlConnector.Protocol.Payloads;
+
+internal interface IColumnReader
```

Comment:
Seems like an `abstract` base class would be more useful here; it could contain the commonly repeated `throw new InvalidCastException` implementation for `GetInt32`.

Additionally, if a (say) `GetInt64` method were added in the future, a common implementation could be added once in the ABC, instead of having to be repeated in each concrete implementation.

Will fix in https://github.com/mysql-net/MySqlConnector/pull/1332/commits/e130a7dc8c28694f23d588c1be40ffca3313c22b.